x-element.js

```diff
@@ -894,39 +892,38 @@ export default class XElement extends HTMLElement {
   }
 
   // Public properties which are serializable or typeless have attributes.
-  static __propertyHasAttribute(property) {
-    return !property.internal && (XElement.__serializableTypes.has(property.type) || !property.type);
+  static #propertyHasAttribute(property) {
+    return !property.internal && (XElement.#serializableTypes.has(property.type) || !property.type);
   }
 
-  static __getTypeName(value) {
+  static #getTypeName(value) {
     return Object.prototype.toString.call(value).slice(8, -1);
   }
 
-  static __notNullish(value) {
+  static #notNullish(value) {
     return value !== undefined && value !== null;
   }
 
-  static __typeIsWrong(type, value) {
+  static #typeIsWrong(type, value) {
     // Because `instanceof` fails on primitives (`'' instanceof String === false`)
     // and `Object.prototype.toString` cannot handle inheritance, we use both.
     return (
-      XElement.__notNullish(value) === false ||
-      (!(value instanceof type) && XElement.__getTypeName(value) !== type.name)
+      XElement.#notNullish(value) === false ||
+      (!(value instanceof type) && XElement.#getTypeName(value) !== type.name)
     );
   }
 
-  static __camelToKebab(camel) {
-    if (XElement.__caseMap.has(camel) === false) {
-      XElement.__caseMap.set(camel, camel.replace(/([A-Z])/g, '-$1').toLowerCase());
+  static #camelToKebab(camel) {
+    if (XElement.#caseMap.has(camel) === false) {
+      XElement.#caseMap.set(camel, camel.replace(/([A-Z])/g, '-$1').toLowerCase());
     }
-    return XElement.__caseMap.get(camel);
+    return XElement.#caseMap.get(camel);
   }
-}
 
-// TODO: #63: define as private class fields inside the constructor when supported.
-XElement.__constructors = new WeakMap();
-XElement.__hosts = new WeakMap();
-XElement.__propertyKeys = new Set(['type', 'attribute', 'input', 'compute', 'observe', 'reflect', 'internal', 'readOnly', 'initial', 'default']);
-XElement.__serializableTypes = new Set([Boolean, String, Number]);
-XElement.__caseMap = new Map();
-XElement.__prototypeInterface = new Set(Object.getOwnPropertyNames(XElement.prototype));
+  static #constructors = new WeakMap();
```

Comment:
Note that these are now defined _within_ the class, rather than outside-and-after. This was always the goal.

Reply:
cooooool